generator/restrictions.cpp

```diff
@@ -0,0 +1,180 @@
+#include "generator/restrictions.hpp"
+
+#include "base/assert.hpp"
+#include "base/logging.hpp"
+#include "base/stl_helpers.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/fstream.hpp"
+#include "std/string.hpp"
+#include "std/vector.hpp"
+
+namespace
+{
+string const kNoStr = "No";
```

Comment:
s/kNoStr/kNo/g
s/kOnlyStr/kOnly/g

BTW, char[] is almost the same but cheaper.